core/src/main/java/org/mule/runtime/core/api/management/stats/AllStatistics.java

```diff
@@ -86,4 +94,36 @@ public synchronized void remove(FlowConstructStatistics stat) {
   public FlowConstructStatistics getApplicationStatistics() {
     return appStats;
   }
+
+  /**
+   * @return the available payload statistics for all components.
+   * @since 4.4, 4.3.1
+   */
+  public Collection<PayloadStatistics> getPayloadStatistics() {
+    return payloadStatistics.values();
+  }
+
+  /**
+   * @param component the component to get the statistics for.
+   * @return the statistics for the provided {@code component}.
+   * @since 4.4, 4.3.1
+   */
+  public PayloadStatistics computePayloadStatisticsIfAbsent(Component component) {
+    return payloadStatistics.computeIfAbsent(component.getLocation().getLocation(),
+                                             loc -> {
+                                               final PayloadStatistics statistics =
+                                                   new PayloadStatistics(loc, component.getIdentifier().toString());
+                                               statistics.setEnabled(isEnabled());
+                                               return statistics;
+                                             });
+  }
+
+  /**
+   * @param componentLocation the location of the component to get the statistics for.
+   * @return the statistics for the component with the provided {@code componentLocation}.
+   * @since 4.4, 4.3.1
+   */
+  public PayloadStatistics getPayloadStatistics(String componentLocation) {
```

Comment:
It should be better to receive a Component instead of a String and do the transformation (`component.getLocation().getLocation()`) inside this method to be consistent with `computePayloadStatisticsIfAbsent`

Reply:
This is on purpose, i don't need this method to be consistent because this method is to be used within ARM, and i don't want to force them to obtain a proper Component, when the may already know the location they want.